django/db/backends/postgresql/introspection.py

```diff
@@ -22,15 +23,45 @@ class DatabaseIntrospection(BaseDatabaseIntrospection):
         700: 'FloatField',
         701: 'FloatField',
         869: 'GenericIPAddressField',
+        1000: ('django.contrib.postgresql.fields.ArrayField',
```

Comment:
I did never think about introspection, but I would imagine something like the admin registry. Third party apps could register a function that takes a *type code* and returns a field instance. Unfortunately here it looks like *type code*s are postgres specific so any such registry API would end up being backend specific? Tricky.